src/transformation/visitors/template.ts

```diff
@@ -65,29 +65,35 @@ export const transformTaggedTemplateExpression: FunctionVisitor<ts.TaggedTemplat
 
     // Construct table with strings and literal strings
 
-    const rawStringsTable = lua.createTableExpression(
-        rawStrings.map(text => lua.createTableFieldExpression(lua.createStringLiteral(text)))
+    const rawStringsObject = ts.factory.createArrayLiteralExpression(
+        rawStrings.map(text => ts.factory.createStringLiteral(text))
     );
 
-    const stringTableLiteral = lua.createTableExpression([
-        ...strings.map(partialString => lua.createTableFieldExpression(lua.createStringLiteral(partialString))),
-        lua.createTableFieldExpression(rawStringsTable, lua.createStringLiteral("raw")),
+    const stringObject = ts.factory.createObjectLiteralExpression([
```

Comment:
Is this really necessary? Shouldnt transforming the array literal already take care of all of this?

Reply:
The object requires both array indices and the "raw" field, so I'm sure there's a way to reuse the array literal to construct this in a single expression.